blockchain/standalone/pow_test.go

```diff
@@ -393,3 +393,1676 @@ func TestCheckProofOfWork(t *testing.T) {
+			{9223372036854775805, 2147483047, 0x1b00a5a6},
+			{9223372036854775806, 2147483347, 0x1b00a5a6},
+			{9223372036854775807, 2147483647, 0x1b00a5a6},
+			{9223372036854775808, 2147483947, 0x1b00a5a6},
+			{9223372036854775809, 2147484247, 0x1b00a5a6},
+		},
+	}, {
+		name: "mainnet: blocks across potential internal calculation overflow" +
+			" boundaries",
+		params:        &mockMainNetParams,
+		startDiffBits: 0x1d00ffff,
+		startHeight:   0,
+		startTime:     0,
+		perInstanceTests: []perInstanceTest{
+			{1, 43200 * 16, 0x1d00ffff},
+			{2, 43200 * 17, 0x1d00ffff},
+			{3, 43200 * 32, 0x1d00ffff},
+			{4, 43200 * 33, 0x1d00ffff},
+			{5, 43200 * 272, 0x1d00ffff},
+			{6, 43200 * 273, 0x1d00ffff},
+		},
+	}, {
+		name:          "testnet: steady 120s blocks at proof of work limit",
+		params:        &mockTestNetParams,
+		startDiffBits: 0x1e00ffff,
+		startHeight:   0,
+		startTime:     0,
+		perInstanceTests: []perInstanceTest{
+			{1, 120, 0x1e00ffff},
+			{2, 240, 0x1e00ffff},
+			{3, 360, 0x1e00ffff},
+			{4, 480, 0x1e00ffff},
+			{5, 600, 0x1e00ffff},
+			{6, 720, 0x1e00ffff},
+			{7, 840, 0x1e00ffff},
+			{8, 960, 0x1e00ffff},
+			{9, 1080, 0x1e00ffff},
+			{10, 1200, 0x1e00ffff},
+		},
+	}, {
+		name: "testnet: random solve times per Poisson distribution for a " +
+			"stable hashrate",
+		params:        &mockTestNetParams,
+		startDiffBits: 0x1c014388,
+		startHeight:   0,
+		startTime:     0,
+		perInstanceTests: []perInstanceTest{
+			{1, 3, 0x1c012119},
+			{2, 34, 0x1c010958},
+			{3, 51, 0x1c00f045},
+			{4, 126, 0x1c00e614},
+			{5, 135, 0x1c00cec4},
+			{6, 431, 0x1c00f4f2},
+			{7, 465, 0x1c00e179},
+			{8, 588, 0x1c00e220},
+			{9, 886, 0x1c010c6e},
+			{10, 936, 0x1c00faeb},
+			{11, 1516, 0x1c0186b8},
+			{12, 1599, 0x1c017910},
+			{13, 1829, 0x1c01a322},
+			{14, 1900, 0x1c018fd6},
+			{15, 2005, 0x1c018a1d},
+			{16, 2106, 0x1c0182fb},
+			{17, 2157, 0x1c016a22},
+			{18, 2160, 0x1c014388},
+		},
+	}, {
+		name:          "simnet: steady 1s blocks at proof of work limit",
+		params:        &mockSimNetParams,
+		startDiffBits: 0x207fffff,
+		startHeight:   0,
+		startTime:     0,
+		perInstanceTests: []perInstanceTest{
+			{1, 1, 0x207fffff},
+			{2, 2, 0x207fffff},
+			{3, 3, 0x207fffff},
+			{4, 4, 0x207fffff},
+			{5, 5, 0x207fffff},
+			{6, 6, 0x207fffff},
+			{7, 7, 0x207fffff},
+			{8, 8, 0x207fffff},
+			{9, 9, 0x207fffff},
+			{10, 10, 0x207fffff},
+		},
+	}, {
+		name: "simnet: random solve times per Poisson distribution for a " +
+			"stable hashrate",
+		params:        &mockSimNetParams,
+		startDiffBits: 0x1e1b4e81,
+		startHeight:   0,
+		startTime:     0,
+		perInstanceTests: []perInstanceTest{
+			{1, 1, 0x1e1b4e81},
+			{2, 3, 0x1e1ea740},
+			{3, 4, 0x1e1ea740},
+			{4, 6, 0x1e226724},
+			{5, 7, 0x1e226724},
+			{6, 7, 0x1e1ea740},
+			{7, 7, 0x1e1b4e81},
+			{8, 8, 0x1e1b4e81},
+			{9, 8, 0x1e185487},
+			{10, 10, 0x1e1b4e81},
+			{11, 10, 0x1e185487},
+			{12, 12, 0x1e1b4e81},
+		},
+	}}
```

Comment:
This is missing a test when both `heightDelta` and `timeDelta` == 0

Reply:
Good call.  Added to `"mainnet: steady 300s blocks at difficulty 0x1b00a5a6"`